pkg/syncer/cnsoperator/manager/init.go

```diff
@@ -87,13 +87,20 @@ func InitCnsOperator(configInfo *types.ConfigInfo) error {
 	crdKindVolumeMetadata := reflect.TypeOf(cnsvolumemetadatav1alpha1.CnsVolumeMetadata{}).Name()
 	crdNameVolumeMetadata := apis.CnsVolumeMetadataPlural + "." + apis.SchemeGroupVersion.Group
 
-	err = k8s.CreateCustomResourceDefinition(ctx, crdNameVolumeMetadata, apis.CnsVolumeMetadataSingular, apis.CnsVolumeMetadataPlural,
+	err = k8s.CreateCustomResourceDefinitionFromSpec(ctx, crdNameVolumeMetadata, apis.CnsVolumeMetadataSingular, apis.CnsVolumeMetadataPlural,
 		crdKindVolumeMetadata, apis.SchemeGroupVersion.Group, apis.SchemeGroupVersion.Version, apiextensionsv1beta1.NamespaceScoped)
 	if err != nil {
 		log.Errorf("failed to create %q CRD. Err: %+v", crdKindVolumeMetadata, err)
 		return err
 	}
 
+	// Create CnsRegisterVolume CRD from manifest
+	err = k8s.CreateCustomResourceDefinitionFromManifest(ctx, "cnsregistervolume_crd.yaml")
```

Comment:
Where is `cnsregistervolume_crd.yaml` located? Do you make it available in the driver container?
Also, why are we creating some CRDs from spec and some from file?

Reply:
The file is located at `pkg/apis/cnsoperator/config/cnsregistervolume_crd.yaml`. Its made available to the syncer container in the docker file - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/images/syncer/Dockerfile#L55

> Also, why are we creating some CRDs from spec and some from file?

I think it's because `CnsRegisterVolume` is a user facing API (created by the admin) while other APIs we've introduced is for internal use. That's my assumption, @BaluDontu can correct me here.

Reply:
No, we want to move all to using manifest. There is a bug tracking this purpose. CnsNodeVmAttachment and CnsVolumeMetadata currently use spec and CnsRegisterVolume uses manifest for now. Thats why we have are multiple ways here. Once we move everything to manifest way, we can remove spec based creation.